### PR TITLE
fixes for 10.13 u3dreal

### PR DESCRIPTION
**Pull request by u3dreal**
it should now be possible to compile luxcore on OSX 10.13 executing....

./cut_deps_release in MacOSCompileDeps and then cmake ../LuxCore in build dir 
and then make_release_osx.sh after copying it over .... might add that stage as well so it would be possible to just run one script with brew deps installed and dylibbuilder installed.

### CODE REVIEW

File: src/luxcore/luxparser/luxparse.y [hunk 1]
```diff
@@ -17,7 +17,7 @@
  *   along with this program.  If not, see <http://www.gnu.org/licenses/>. *
  ***************************************************************************/
 
-%name-prefix "luxcore_parserlxs_yy"
+%name-prefix = "luxcore_parserlxs_yy"
```
Comment (Dade916) [inline src/luxcore/luxparser/luxparse.y]: Can you check if replacing:

`%name-prefix "luxcore_parserlxs_yy"`

with:

`%define api.prefix {luxcore_parserlxs_yy}`

work for you ? It is apparently the right and latest way to declare the prefix.

Comment (u3dreal) [reply src/luxcore/luxparser/luxparse.y]: suggestion 1 works 

in luxparse.y i get error
/Users/drquader/Documents/GitHub/LuxCore/src/luxcore/luxparser/luxparse.y:20.9-18: syntax error, unexpected identifier, expecting string

File: src/luxrays/core/context.cpp [hunk 2]
```diff
@@ -49,8 +49,8 @@ Context::Context(LuxRaysDebugHandler handler, const Properties &config) : cfg(co
 	} catch (cl::Error &err) {
 		// The cl_khr_icd extension throws exceptions if zero platforms are available.
 		// We ignore that error (OpenCL is optional), but throw anything else.
-		if (err.err() != CL_PLATFORM_NOT_FOUND_KHR)
-			throw;
+		//if (err.err() != CL_PLATFORM_NOT_FOUND_KHR)
+		//	throw;
```
[the remark this answers is not in the document]
Comment (u3dreal) [reply src/luxrays/core/context.cpp]: OK work like this

#if defined(cl_khr_idc)
		if (err.err() != CL_PLATFORM_NOT_FOUND_KHR)
			throw;
    #endif

File: src/luxcore/luxparser/luxparse.y (hunk 1 above)
Comment (u3dreal) [reply src/luxcore/luxparser/luxparse.y]: Hmm it's related to bison right ??

i found 

// Prefix all tokens
%define api.token.prefix {SCHEMA_}

Comment (u3dreal) [reply src/luxcore/luxparser/luxparse.y]: https://twitter.com/plusvic/status/960997335835824128

maybe use newer bison ? what version do you use ??

Comment (u3dreal) [reply src/luxcore/luxparser/luxparse.y]: what about a simple 
IF(APPLE) ? is this accessable there ??

%name-prefix = "luxcore_parserlxs_yy"

is correct for bison 2.3 that comes with OSX10.13

Comment (u3dreal) [reply src/luxcore/luxparser/luxparse.y]: using bison 3.4 from brew cellar works but says deprecated ..... but this is not a good solution ...

Comment (u3dreal) [reply src/luxcore/luxparser/luxparse.y]: Does not seem to be as trivial in bison :(

Comment (u3dreal) [reply src/luxcore/luxparser/luxparse.y]: Nevermind a

echo 'export PATH="/usr/local/opt/bison/bin:$PATH"' >> ~/.bash_profile

did the trick

**Comment by Dade916** [comment]
The pull request now includes a lot of unrelated stuff (all my yesterday commits). You need to remove them or it will do a mess.

**Comment by u3dreal** [comment]
Oh no ..... but 

@u3dreal
Revert "Merge remote-tracking branch 'upstream/master'" …
c49f4c3

should have done it 

How do i get updated master ... after merge ??

**Comment by Dade916** [comment]
Click up-right on "File changed" tab and you will see all the modified files by the pull request.

**Comment by u3dreal** [comment]
Sorry it's my first time :) So how do i revert ?? doing itlocally and then puching does not work

**Comment by u3dreal** [comment]
basicly it's only 3 files ... should i close this pull request and do it all over ?? Sorry not really used to all this

**Comment by u3dreal** [comment]
Now it should be OK only 3 files to update and they are correct  ... After that i can do the merge from upstream ???? Sorry for messing things up

**Comment by Theverat** [comment]
No problem, thanks for your contributions!

**Pull request merged**